src/cargo/core/compiler/compile_kind.rs

```diff
@@ -0,0 +1,90 @@
+use crate::core::compiler::BuildContext;
+use crate::core::{InternedString, Target};
+use crate::util::errors::{CargoResult, CargoResultExt};
+use serde::Serialize;
+use std::path::Path;
+
+/// Indicates whether an object is for the host architcture or the target architecture.
+///
+/// These will be the same unless cross-compiling.
```

Comment:
This sentence is a little vague.  I personally would just remove it.  Or an alternative would be to say something like "If not cross-compiling, all units are `CompileKind::Host`."